packages/opentelemetry-plugin-xml-http-request/src/xhr.ts

```diff
@@ -0,0 +1,371 @@
+/*!
+ * Copyright 2019, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  B3Format,
+  BasePlugin,
+  isUrlIgnored,
+  isWrapped,
+  otperformance,
+} from '@opentelemetry/core';
+import * as tracing from '@opentelemetry/tracing';
+import * as types from '@opentelemetry/types';
+import {
+  addSpanNetworkEvent,
+  getResource,
+  parseUrl,
+  PerformanceTimingNames as PTN,
+  sameOriginOrUrlMatches,
+} from '@opentelemetry/web';
+import * as shimmer from 'shimmer';
+import { AttributeNames } from './enums/AttributeNames';
+import { EventNames } from './enums/EventNames';
+import {
+  OpenFunction,
+  PropagateTraceHeaderUrls,
+  SendBody,
+  SendFunction,
+  XMLHttpRequestWrapped,
+} from './types';
+
+/**
+ * XMLHttpRequest config
+ */
+export interface XMLHttpRequestPluginConfig extends types.PluginConfig {
+  // urls which should include trace headers when origin matches
+  propagateTraceHeaderUrls?: PropagateTraceHeaderUrls;
+}
+
+/**
+ * This class represents a XMLHttpRequest plugin for auto instrumentation
+ */
+export class XMLHttpRequestPlugin extends BasePlugin<XMLHttpRequest> {
+  readonly component: string = 'xml-http-request';
+  readonly version: string = '0.3.0';
```

Comment:
Since we have all versions locked together via lerna, can this just import a `VERSION` constant from core? That way it would bump up automatically.

Reply:
That would assume that the user has the same version of the plugin and of core installed which may not always be the case. @obecny and I had talked about adding the autogenerated VERSION file to all lerna packages, but nothing has been done yet.

Reply:
Makes sense. Could we just have this as a todo comment and create an issue for it?